lib/concurrent-ruby/concurrent/atomic/lock_local_var.rb

```diff
@@ -0,0 +1,20 @@
+require_relative 'fiber_local_var'
+require_relative 'thread_local_var'
+
+module Concurrent
+  def self.mutex_owned_per_thread?
+    mutex = Mutex.new
+
+    # Lock the mutex:
+    mutex.synchronize do
+      # Check if the mutex is still owned in a child fiber:
+      Fiber.new{mutex.owned?}.resume
+    end
+  end
+
+  if mutex_owned_per_thread?
+    LockLocalVar = ThreadLocalVar
+  else
+    LockLocalVar = FiberLocalVar
+  end
```

Comment:
I'm unsure about this name, especially a public constant like this under `Concurrent`.
I think for now it might be better to just use that logic in ReentrantReadWriteLock.

Reply:
Feel free to suggest a better name?

I am happy to move this logic but it seems like external consumers might run into the same problem and thus exposing it seems pretty reasonable to me. i.e. we use it in the tests too.

Reply:
It seems reasonable to expose it and low enough risk.
Could be MutexLocalVar but it also applies to Monitor, so LockLocalVar sounds fine.